R/preprocessing.R

```diff
@@ -500,39 +500,91 @@ GetResidual <- function(
 #' }
 #'
 Load10X_Spatial <- function(
-  data.dir,
-  filename = 'filtered_feature_bc_matrix.h5',
-  assay = 'Spatial',
-  slice = 'slice1',
-  filter.matrix = TRUE,
-  to.upper = FALSE,
-  image = NULL,
-  ...
+    data.dir,
+    filename = 'filtered_feature_bc_matrix.h5',
+    assay = 'Spatial',
+    slice = 'slice1',
+    filter.matrix = TRUE,
+    to.upper = FALSE,
+    image = NULL,
+    ...
 ) {
   if (length(x = data.dir) > 1) {
-    warning("'Load10X_Spatial' accepts only one 'data.dir'", immediate. = TRUE)
+    warning("'Load10X_Spatial' accepts only one 'data.dir'",
+            immediate. = TRUE)
     data.dir <- data.dir[1]
   }
-  data <- Read10X_h5(filename = file.path(data.dir, filename), ...)
+  data <- Read10X_h5(filename = file.path(data.dir, filename),
+                     ...)
+
   if (to.upper) {
-    rownames(x = data) <- toupper(x = rownames(x = data))
+    data <- imap(data, ~{
+      rownames(.x) <- toupper(rownames(.x))
+      .x
+    })
+  }
+  if (is.list(data) && "Antibody Capture" %in% names(data)) {
+    matrix_gex <- data$`Gene Expression`
+    matrix_protein <- data$`Antibody Capture`
+    object <- CreateSeuratObject(counts = matrix_gex, assay = assay)
+    object_protein <- CreateAssayObject(counts = matrix_protein)
+    object[["Protein"]] <- object_protein
+  }
+  else {
+    object <- CreateSeuratObject(counts = data, assay = assay)
   }
-  object <- CreateSeuratObject(counts = data, assay = assay)
   if (is.null(x = image)) {
-    image <- Read10X_Image(
-	    image.dir = file.path(data.dir, 'spatial'),
-	    filter.matrix = filter.matrix
-  	)
-  } else {
+    image <- Read10X_Image(image.dir = file.path(data.dir,"spatial"),
+                           filter.matrix = filter.matrix)
+  }
+  else {
     if (!inherits(x = image, what = "VisiumV1"))
       stop("Image must be an object of class 'VisiumV1'.")
   }
   image <- image[Cells(x = object)]
   DefaultAssay(object = image) <- assay
   object[[slice]] <- image
+
+  # if using the meta-data available for probes add to @misc slot
+  file_path <- file.path(data.dir, filename)
+  infile <- hdf5r::H5File$new(filename = file_path, mode = 'r')
+  if("matrix/features/probe_region" %in% hdf5r::list.objects(infile)) {
+    probe_metadata <- Read10x_probe_metadata(data.dir)
+    Misc(object = object[['Spatial']], slot = "probe_metadata") <- probe_metadata
+  }
```

Comment:
Hey @stephenwilliams22, I'm having a hard time understanding these lines. It seem like with `filename` as the default value, the probe info won't be loaded since `file_path` will point to the filtered h5. Should there be a separate variable (like `probes.filename`) for `raw_probe_bc_matrix.h5` which contains the probe information (and also 'matrix/features/probe_region')? Let me know what you think or if I'm misunderstanding

Reply:
Hey @AustinHartman. I actually wrote it this way if we ever decided to include the probe metadata in the raw or filtered feature_bc_matrix.h5. The way the code is written if you don't use `filename=raw_probe_bc_matrix.h5` `if("matrix/features/probe_region" %in% hdf5r::list.objects(infile))` will catch that that slot isn't there and skip making the meta-data. if you'd rather I update the code to not  run ```file_path <- file.path(data.dir, filename)
  infile <- hdf5r::H5File$new(filename = file_path, mode = 'r')``` I'm happy to do that as well. As it stands the code works for `raw_probe_bc_matrix.h5`, `raw_feature_bc_matrix.h5`, and `filtered_feature_bc_matrix.h5` and will only write probe_metadata for `raw_probe_bc_matrix.h5`

Reply:
Got it. In that case, do you think `filename` should be passed to `Read10x_probe_metadata` in the event that the `raw_probe_bc_matrix.h5` has a different name? It seems possible that the `if` condition passes, but `Read10x_probe_metadata` could fail with `File not found`.

Reply:
Yes, that's a good idea. I'll make the fix